### PR TITLE
Attempt to run Travis CI builds on both OS X and Linux.

### PR DESCRIPTION
**Pull request by mdickinson**
Could it possibly be this easy? Probably not, but let's find out.

### CODE REVIEW

**Comment by itziakos** [comment]
travis osx does not know about python so you probably need to install it https://github.com/enthought/mayavi/blob/master/travis/before_install.sh

**Comment by mdickinson** [comment]
Nope, not that easy.

Output from the [Python 2.7 OS X build](https://travis-ci.org/enthought/traits/jobs/135900878):
```
Worker information
hostname: worker-jupiter-brain:72ac8ddf-55ed-487e-884f-437d525e126c
version: v2.0.0 https://github.com/travis-ci/worker/tree/ca6cb0c5d3920912b1c3acc87c44a5da2120a971
instance: b9964562-8822-47e4-afe8-6ad97010abea:
startup: 1m0.984255511s
Build system information
Build language: python
Build group: stable
Build dist: precise

$ export DEBIAN_FRONTEND=noninteractive
Fix WWDRCA Certificate
2.7 is not installed; attempting download
tar: Unrecognized archive format
tar: Error exit delayed from previous errors.

The command "sudo tar xjf python-2.7.tar.bz2 --directory /" failed and exited with 1 during .

Your build has been stopped.
```

**Comment by mdickinson** [comment]
@itziakos: Indeed; thanks. It looks like using `brew` to install Python is the simplest option here. I'll have a go at that tomorrow.

File: .travis.yml [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
   only:
     master
-cache:
-  directories:
-    - $HOME/.cache
-    - $HOME/.ccache
 before_install:
-  - ccache -s
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install python 3; fi
```
Comment (mdickinson) [inline .travis.yml]: `python3`, not `python 3`.

**Comment by mdickinson** [comment]
Closing. This is redundant with #420.

**Pull request closed without merging**